app/models/manageiq/providers/vmware/infra_manager/provision/configuration/disk.rb

```diff
@@ -21,6 +21,37 @@ def build_config_disk_spec(vmcs)
     end
   end
 
+  def build_disk_relocate_spec(_clone_options)
+    disks_info.map do |_disk|
+      VimHash.new("VirtualDiskFlatVer2BackingInfo") do |vdbi|
+        case get_option(:disk_format)
+        when 'thin'
+          vdbi.thinProvisioned = true # VimString.new('sparse', "VirtualMachineRelocateTransformation")
+        when 'thick'
+          vdbi.thinProvisioned = false
+          vdbi.eagerlyScrub = false # then VimString.new('flat', "VirtualMachineRelocateTransformation")
+        when 'thick_eager'
+          vdbi.eagerlyScrub = true
+          vdbi.thinProvisioned = false
+        end
+      end
+    end
+  end
+
+  def disks_info
+    inventory_hash = source.with_provider_connection do |vim|
```

Comment:
Once we get this working we should probably cache the device list so that we only fetch it once, something like

```
def source_vm_devices
  @source_vm_devices ||= begin
    source.with_provider_connection { |vim| vim.virtualMachineByMor().dig("config", "hardware", "device") || []
  end
end
```